src/communication/mpi.c

```diff
@@ -132,73 +114,46 @@ void send_remote_msg(msg_t *msg){
  * This function is thread-safe.
  */
 void receive_remote_msgs(void){
-	int res = 0;
 	int size;
 	msg_t *msg;
 	MPI_Status status;
-	bool found_msg = true;
+	MPI_Message mpi_msg;
+	int pending;
 
-	/* - `pending_msgs` and `MPI_Recv` need to be in the same critical section.
-	 *    I could start an MPI_Recv with an empty incoming queue.
-	 * - `MPI_Recv` and `insert_bottom_half` need to be in the same critical section.
-	 *    messages need to be inserted in arrival order into the BH
-	 */
 	if(!spin_trylock(&msgs_lock))
```

Comment:
```suggestion
	if(!spin_trylock(&msgs_lock))
```

i think we could remove this, we tried to run without the lock and everything was working

Reply:
We tried as well, and it's actually crashing. We can remove the spinlock, but this depends on #102